rust/template/channel.rs

```diff
@@ -0,0 +1,32 @@
+use fallible_iterator::FallibleIterator;
+use differential_datalog::record::Record;
+use differential_datalog::program::Response;
+
+// The consumer can subscribe to the channel
+// which acts as an observable of deltas.
+pub trait Observable<T, E>
+{
+    fn subscribe<'a>(&'a mut self, observer: Box<dyn Observer<T, E>>) -> Box<dyn Subscription + 'a>;
+}
+
+// The channel is an observer of changes from
+// a producer
+pub trait Observer<T, E>
+{
+    fn on_start(&self) -> Response<()>;
```

Comment:
How is the producer supposed to handle errors from this interface? E.g., should they stop sending data?

Reply:
There's another method `on_error` from `Observer` that I'll write once the other parts come into shape.

Reply:
Added `on_error` to `Observer`.